WordPress/gutenberg
# List View: add an indicator of when a position type is set for a block

## What?


Part of https://github.com/WordPress/gutenberg/issues/47043

Add an indicator to the block icon in the list view to flag a position type when one is set.

This follows on from feedback in https://github.com/WordPress/gutenberg/pull/49085#issuecomment-1469711661

## Why?


When a user sets a Group to sticky, it would be helpful for the sticky state to be visible from within the list view.

## How?


* Add a function to get a position type label based on a block's attributes.
* Update the `useBlockDisplayInformation` hook to retrieve and pass along the position label and position type.
* Add a `pinSmall` icon.
* When a block's position is set to `sticky`, display a pin icon next to where the lock icon is displayed.



## Testing Instructions





1. With TT3 active (or any blocks theme with `appearanceTools` set to true), add a Group block at the root level of a post, or at the root level of a template.
2. In the sidebar Position panel, select "Sticky".
3. In the List view, you should see a pin icon flagging that the block is sticky.

## Screenshots or screencast 

<img width="347" alt="image" src="https://github.com/WordPress/gutenberg/assets/14988353/2fd5e59b-50a5-49ab-b3d7-1f35b1fe4b16">
